src/writer.js

```diff
@@ -0,0 +1,377 @@
+            setTimeout(function () {
+              showMsg(`<b>Congratulations! <br> New Record! <br></b>You paired all the cards in ${time} and ${turns} turns <br>`);
+              document.getElementsByClassName('msgDiv')[0].append(createHtml('button', 'msgButton', 'Play Again'));
+              document.getElementsByClassName('msgButton')[0].addEventListener('click', changeFieldSize);
+            }, 300);
+            changeLeadResult(result);
+          }
+          else {
+            setTimeout(function () {
+              showMsg(`<b>Good, But You Gotta Try Harder To Beat The Record! <br> </b>You paired all the cards in ${time} and ${turns} turns <br>`);
+              document.getElementsByClassName('msgDiv')[0].append(createHtml('button', 'msgButton', 'Play Again'));
+              document.getElementsByClassName('msgButton')[0].addEventListener('click', changeFieldSize);
+            }, 300);
+          }
+
+          startStop();
+          stopTurns();
+          emptyCardNumber = 0;
+        }
+
+        faceUpCounter = 0;
+        return;
+      }
+      faceUpCard1 = target;
+    }
+  }
+};
+
+export function isFirstBetter (recent, previous) {
+  let recentTime = Number(recent.time.replace(':', ''));
+  let recentRatio = recentTime + recent.turns;
+
+  //make recent better if there is no previous
+  if (!localStorage.getItem(recent.size)) {
+    return 1;
+  }
+
+  let previousTime = Number(JSON.parse(previous).time.replace(':', ''));
+  let previousRatio = previousTime + JSON.parse(previous).turns;
+
+  //if new result is worse than saved one
+  if (recentRatio >= previousRatio) {
+    return 0;
+  }
+
+  //if new result is better than saved one
+  return 1;
+}
+
+export function changeLeadResult (result) {
+  localStorage.removeItem(result.size);
+  document.getElementsByClassName(`size${result.size}`)[0].innerHTML = `<b>Time:</b> ${result.time}, <b>Turns:</b> ${result.turns} `;
+  localStorage.setItem(result.size, JSON.stringify(result));
+}
+
+export function showLeaderboard () {
+  if (localStorage.getItem(44)) {
+    let res1 = JSON.parse(localStorage.getItem(44));
+    document.getElementsByClassName('size44')[0].innerHTML = `<b>Time:</b> ${res1.time}, <b>Turns:</b> ${res1.turns} `;
+  }
+  if (localStorage.getItem(54)) {
+    let res2 = JSON.parse(localStorage.getItem(54));
+    document.getElementsByClassName('size54')[0].innerHTML = `<b>Time:</b> ${res2.time}, <b>Turns:</b> ${res2.turns} `;
+  }
+  if (localStorage.getItem(66)) {
+    let res3 = JSON.parse(localStorage.getItem(66));
+    document.getElementsByClassName('size66')[0].innerHTML = `<b>Time:</b> ${res3.time}, <b>Turns:</b> ${res3.turns} `;
+  }
+}
+
+export function saveGame (time, turns) {
+  let saveTime = document.getElementsByClassName('stopwatch')[0].value;
+  let saveTurns = document.getElementsByClassName('turnsCounter')[0].value;
+  let saveFieldSize = document.getElementsByClassName('fieldSelector')[0].value;
```

Comment:
i don'y like you serach elements each time you do interaction. maybe it is much better to save elements to variables during initialization.

Reply:
Done where it was needed and possible